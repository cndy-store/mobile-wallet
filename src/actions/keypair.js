import { getItem, setItem, removeItem } from '../lib/keypairStorage';
import { StellarSdk } from '../lib/stellar';
const { Keypair, StrKey } = StellarSdk;

const STORAGE_KEY = 'STELLAR_SECRET';

export const KEYPAIR_STARTS_PROCESSING = 'KEYPAIR_STARTS_PROCESSING';
export const KEYPAIR_ERROR = 'KEYPAIR_ERROR';
export const KEYPAIR_LOAD = 'KEYPAIR_LOAD';
export const KEYPAIR_SAVE = 'KEYPAIR_SAVE';
export const KEYPAIR_REMOVE = 'KEYPAIR_REMOVE';

const chomp = str => str.replace(/\n|\r| /g, '');
const isValidSecret = secret => StrKey.isValidEd25519SecretSeed(secret);

export const keypairProcessing = bool => ({
  type: KEYPAIR_STARTS_PROCESSING,
  isProcessing: bool
});

export const keypairErrored = ({ error }) => ({
  type: KEYPAIR_ERROR,
  error
});

export const keypairLoadSucceded = ({ secret }) => {
  const keypair = secret ? Keypair.fromSecret(secret) : null;

  return {
    type: KEYPAIR_LOAD,
    keypair: keypair,
    error: null
  };
};

export const keypairSaveSucceded = ({ secret }) => {
  const keypair = Keypair.fromSecret(secret);

  return {
    type: KEYPAIR_SAVE,
    keypair: keypair,
    error: null
  };
};

export const keypairRemoveSucceded = () => ({
  type: KEYPAIR_REMOVE,
  keypair: null,
  error: null
});

export const loadKeypair = () => {
  return async dispatch => {
    dispatch(keypairProcessing(true));
    try {
      const secret = await getItem(STORAGE_KEY);
      dispatch(keypairProcessing(false));
      dispatch(keypairLoadSucceded({ secret }));
      return Promise.resolve({ secret });
    } catch (error) {
      dispatch(keypairProcessing(false));
      dispatch(keypairErrored({ error }));
      return Promise.reject({ error });
    }
  };
};

export const saveKeypair = secret => {
  secret = chomp(secret);

  return async dispatch => {
    dispatch(keypairProcessing(true));

    if (!isValidSecret(secret)) {
      const error = new Error('Invalid Secret');
      dispatch(keypairProcessing(false));
      dispatch(keypairErrored({ error }));
      return Promise.reject({ error });
    }

    try {
      await setItem(STORAGE_KEY, secret);
      dispatch(keypairProcessing(false));
      dispatch(keypairSaveSucceded({ secret }));
      return Promise.resolve({ secret });
    } catch (error) {
      dispatch(keypairProcessing(false));
      dispatch(keypairErrored({ error }));
      return Promise.reject({ error });
    }
  };
};

export const removeKeypair = () => {
  return async dispatch => {
    dispatch(keypairProcessing(true));

    try {
      await removeItem(STORAGE_KEY);
      dispatch(keypairProcessing(false));
      dispatch(keypairRemoveSucceded());
      return Promise.resolve();
    } catch (error) {
      dispatch(keypairProcessing(false));
      dispatch(keypairErrored({ error }));
      return Promise.reject({ error });
    }
  };
};
