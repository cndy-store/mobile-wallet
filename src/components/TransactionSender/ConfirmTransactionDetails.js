import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';

import {
  Icon,
  H1,
  Body,
  Card,
  CardItem,
  Container,
  Header,
  Left,
  Right,
  Content,
  Button,
  Text,
  Title
} from 'native-base';
import Dimensions from 'Dimensions';
import ShortenedPublicKey from '../../components/ShortenedPublicKey';
import { sanitizeOngoingAmountInput } from '../../lib/formatter';

class ConfirmTransactionDetails extends Component {
  constructor(props) {
    super(props);

    this.handleConfirm = this.handleConfirm.bind(this);
    this.handleReject = this.handleReject.bind(this);
  }

  handleConfirm() {
    this.props.onConfirm();
  }

  handleReject() {
    this.props.onReject();
  }

  render() {
    return (
      <View>
        <Card>
          <CardItem header>
            <Text>Confirm Transaction Details</Text>
          </CardItem>
          <CardItem />
          <CardItem>
            <Text>Receiver</Text>
          </CardItem>
          <CardItem>
            <ShortenedPublicKey publicKey={this.props.receiver} />
          </CardItem>
          <CardItem>
            <Text>Amount</Text>
          </CardItem>
          <CardItem>
            <H1>{this.props.amount}</H1>
          </CardItem>
          <CardItem>
            <Body>
              <Button block onPress={this.handleConfirm}>
                <Text>Confirm</Text>
              </Button>
            </Body>
          </CardItem>
        </Card>

        <Button block transparent onPress={this.handleReject}>
          <Text>Change amount</Text>
        </Button>
      </View>
    );
  }
}

ConfirmTransactionDetails.propTypes = {
  receiver: PropTypes.string.isRequired,
  amount: PropTypes.string.isRequired,
  onConfirm: PropTypes.func.isRequired,
  onReject: PropTypes.func.isRequired
};

export default ConfirmTransactionDetails;
