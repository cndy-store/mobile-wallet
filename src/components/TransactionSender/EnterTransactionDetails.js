import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  Button,
  StatusBar,
  StyleSheet,
  View,
  Text,
  TextInput
} from 'react-native';
import Dimensions from 'Dimensions';
import ShortenedPublicKey from '../../components/ShortenedPublicKey';
import { sanitizeOngoingAmountInput } from '../../lib/formatter';

const { width, height } = Dimensions.get('window');

class EnterTransactionDetails extends Component {
  constructor(props) {
    super(props);

    this.state = {
      amountInput: ''
    };

    this.updateAmount = this.updateAmount.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  updateAmount(amountInput) {
    this.setState({ amountInput: sanitizeOngoingAmountInput(amountInput) });
  }

  handleSubmit() {
    this.props.onSubmit({ amountInput: this.state.amountInput });
  }

  renderError() {
    if (!this.props.error) return null;

    return <Text>{this.props.error}</Text>;
  }

  render() {
    return (
      <View>
        <ShortenedPublicKey publicKey={this.props.receiver} />
        <Text>Please enter your Amount</Text>
        <TextInput
          style={{
            width,
            height: 40,
            borderColor: 'gray',
            borderWidth: 1
          }}
          value={this.state.amountInput}
          autoCorrect={false}
          keyboardType={'numeric'}
          onChangeText={this.updateAmount}
        />
        <Button title="Next" onPress={this.handleSubmit} />
        {this.renderError()}
      </View>
    );
  }
}

EnterTransactionDetails.propTypes = {
  receiver: PropTypes.string.isRequired,
  error: PropTypes.string,
  onSubmit: PropTypes.func.isRequired
};

export default EnterTransactionDetails;
