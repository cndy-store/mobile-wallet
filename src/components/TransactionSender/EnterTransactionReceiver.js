import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { StatusBar, View } from 'react-native';
import {
  Icon,
  Body,
  Card,
  CardItem,
  Container,
  Form,
  Input,
  Label,
  Header,
  Item,
  Left,
  Right,
  Content,
  Button,
  Text,
  Title
} from 'native-base';
import { styles as s } from 'react-native-style-tachyons';
import FormError from '../../components/FormError';
import Dimensions from 'Dimensions';

const { width, height } = Dimensions.get('window');

class EnterTransactionReceiver extends Component {
  constructor(props) {
    super(props);

    this.state = {
      receiverInput: ''
    };

    this.updateReceiver = this.updateReceiver.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  updateReceiver(receiverInput) {
    this.setState({ receiverInput });
  }

  handleSubmit() {
    this.props.onSubmit({ receiverInput: this.state.receiverInput });
  }

  renderError() {
    if (!this.props.error) return null;

    return <FormError message={this.props.error} />;
  }

  render() {
    return (
      <View>
        <Card>
          <CardItem header>
            <Text>Enter Public Key of Receiver</Text>
          </CardItem>
          <CardItem>
            <View style={{ flex: 1 }}>
              <Form>
                <Item stackedLabel error={!!this.state.error}>
                  <Label>Receiver</Label>
                  <Input
                    autoFocus
                    value={this.state.receiverInput}
                    autoCorrect={false}
                    onChangeText={this.updateReceiver}
                  />
                </Item>
                {this.renderError()}
              </Form>
            </View>
          </CardItem>
          <CardItem>
            <Body>
              <Button block onPress={this.handleSubmit}>
                <Text>Next</Text>
              </Button>
            </Body>
          </CardItem>
        </Card>
      </View>
    );
  }
}

EnterTransactionReceiver.propTypes = {
  error: PropTypes.string,
  onSubmit: PropTypes.func.isRequired
};

export default EnterTransactionReceiver;
