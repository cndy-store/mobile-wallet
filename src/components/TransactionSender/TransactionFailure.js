import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { StatusBar, View } from 'react-native';
import {
  Icon,
  H1,
  Body,
  Card,
  CardItem,
  Container,
  Form,
  Input,
  Label,
  Header,
  Item,
  Left,
  Right,
  Content,
  Button,
  Text,
  Title
} from 'native-base';
import { styles as s } from 'react-native-style-tachyons';
import ShortenedPublicKey from '../../components/ShortenedPublicKey';

class TransactionFailure extends Component {
  constructor(props) {
    super(props);

    this.handleAcknowledge = this.handleAcknowledge.bind(this);
  }

  handleAcknowledge() {
    this.props.onAcknowledge();
  }

  render() {
    return (
      <View>
        <Card>
          <CardItem header>
            <Text>Transaction Failed!</Text>
          </CardItem>
          <CardItem>
            <Text>
              The transaction could not be completed, please check your
              transaction logs to make sure it took place or retry later.
            </Text>
            <Text>Reason: {this.props.error}</Text>
          </CardItem>
          <CardItem>
            <Text>Receiver</Text>
          </CardItem>
          <CardItem>
            <ShortenedPublicKey publicKey={this.props.receiver} />
          </CardItem>
          <CardItem>
            <Text>Amount</Text>
          </CardItem>
          <CardItem>
            <H1>{this.props.amount}</H1>
          </CardItem>
          <CardItem>
            <Body>
              <Button block danger onPress={this.handleAcknowledge}>
                <Text>Done</Text>
              </Button>
            </Body>
          </CardItem>
        </Card>
      </View>
    );
  }
}

TransactionFailure.propTypes = {
  onAcknowledge: PropTypes.func.isRequired,
  error: PropTypes.string.isRequired
};

export default TransactionFailure;
