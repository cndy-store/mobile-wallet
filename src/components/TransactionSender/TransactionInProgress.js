import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  Button,
  StatusBar,
  StyleSheet,
  View,
  Text,
  TextInput
} from 'react-native';
import Dimensions from 'Dimensions';

const { width, height } = Dimensions.get('window');

class TransactionInProgress extends Component {
  render() {
    return (
      <View>
        <Text>In Progress...</Text>
      </View>
    );
  }
}

TransactionInProgress.propTypes = {};

export default TransactionInProgress;
