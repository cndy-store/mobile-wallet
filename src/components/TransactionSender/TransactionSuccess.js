import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  Button,
  StatusBar,
  StyleSheet,
  View,
  Text,
  TextInput
} from 'react-native';

class TransactionSuccess extends Component {
  render() {
    return (
      <View>
        <Text>Success!</Text>
      </View>
    );
  }
}

TransactionSuccess.propTypes = {};

export default TransactionSuccess;
