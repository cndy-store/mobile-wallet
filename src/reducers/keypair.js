import {
  KEYPAIR_STARTS_PROCESSING,
  KEYPAIR_ERROR,
  KEYPAIR_LOAD,
  KEYPAIR_SAVE,
  KEYPAIR_REMOVE
} from '../actions/keypair';

const defaultState = {
  isProcessing: false,
  error: null,
  keypair: null
};

export default function keypair(state = defaultState, action) {
  switch (action.type) {
    case KEYPAIR_STARTS_PROCESSING:
      return { ...state, isProcessing: action.isProcessing };
    case KEYPAIR_ERROR:
      return {
        ...state,
        error: action.error,
        isProcessing: action.isProcessing
      };
    case KEYPAIR_LOAD:
    case KEYPAIR_SAVE:
    case KEYPAIR_REMOVE:
      return {
        ...state,
        keypair: action.keypair,
        error: action.error,
        isProcessing: action.isProcessing
      };
    default:
      return state;
  }
}
