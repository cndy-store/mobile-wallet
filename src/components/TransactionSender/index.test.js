import React from 'react';
import { render } from '../../__tests__/renderer';
import { publicKey } from '../../__tests__/fixtures/keypair';
import { TransactionSender } from './index';
import ConfirmTransactionDetails from './ConfirmTransactionDetails';
import EnterTransactionReceiver from './EnterTransactionReceiver';
import EnterTransactionDetails from './EnterTransactionDetails';
import TransactionFailure from './TransactionFailure';
import TransactionInProgress from './TransactionInProgress';
import TransactionSuccess from './TransactionSuccess';

const onCancel = jest.fn();
const onSuccess = jest.fn();
const onFailure = jest.fn();
const fakeResponse = { data: 'TRANSACTION' };

it('renders correctly without an error', () => {
  const { toJSON } = render(
    <TransactionSender
      onCancel={onCancel}
      onSuccess={onSuccess}
      onFailure={onFailure}
    />
  );
  expect(toJSON()).toMatchSnapshot();
});

describe('no receiver given', () => {
  const { instance, root } = render(
    <TransactionSender
      onCancel={onCancel}
      onSuccess={onSuccess}
      onFailure={onFailure}
    />
  );

  it('renders EnterTransactionReceiver', () => {
    expect(() => {
      root.findByType(EnterTransactionReceiver);
    }).not.toThrowError();
  });

  it('shows the close button', () => {
    const closeButton = root.findByProps({ title: 'Close' });
    expect(closeButton).toBeDefined();
  });

  it('passes the correct props to the subcomponent', () => {
    instance.setState({ receiverError: 'Some error' });
    const subcomponent = root.findByType(EnterTransactionReceiver);
    expect(subcomponent.props.onSubmit).toEqual(instance.handleReceiverUpdate);
    expect(subcomponent.props.error).toEqual(instance.state.receiverError);
  });
});

describe('no amount given', () => {
  const { instance, root } = render(
    <TransactionSender
      onCancel={onCancel}
      onSuccess={onSuccess}
      onFailure={onFailure}
      receiver={publicKey}
    />
  );

  it('renders EnterTransactionDetails', () => {
    expect(() => {
      root.findByType(EnterTransactionDetails);
    }).not.toThrowError();
  });

  it('shows the close button', () => {
    const closeButton = root.findByProps({ title: 'Close' });
    expect(closeButton).toBeDefined();
  });

  it('passes the correct props to the subcomponent', () => {
    instance.setState({ amountError: 'Some error' });
    const subcomponent = root.findByType(EnterTransactionDetails);
    expect(subcomponent.props.onSubmit).toEqual(instance.handleAmountUpdate);
    expect(subcomponent.props.receiver).toEqual(instance.state.receiver);
    expect(subcomponent.props.error).toEqual(instance.state.amountError);
  });
});

describe('user input complete', () => {
  const { instance, root } = render(
    <TransactionSender
      onCancel={onCancel}
      onSuccess={onSuccess}
      onFailure={onFailure}
      receiver={publicKey}
      amount={'100.00'}
    />
  );

  it('renders ConfirmTransactionDetails', () => {
    expect(() => {
      root.findByType(ConfirmTransactionDetails);
    }).not.toThrowError();
  });

  it('shows the close button', () => {
    const closeButton = root.findByProps({ title: 'Close' });
    expect(closeButton).toBeDefined();
  });

  it('passes the correct props to the subcomponent', () => {
    const subcomponent = root.findByType(ConfirmTransactionDetails);
    expect(subcomponent.props.onConfirm).toEqual(instance.handleConfirmation);
    expect(subcomponent.props.onReject).toEqual(instance.handleRejection);
    expect(subcomponent.props.receiver).toEqual(instance.state.receiver);
    expect(subcomponent.props.amount).toEqual(instance.state.amount);
  });
});

describe('transaction in progress', () => {
  const { instance, root } = render(
    <TransactionSender
      onCancel={onCancel}
      onSuccess={onSuccess}
      onFailure={onFailure}
      receiver={publicKey}
      amount={'100.00'}
    />
  );

  it('renders TransactionInProgress', () => {
    instance.setState({ inProgress: true });

    expect(() => {
      root.findByType(TransactionInProgress);
    }).not.toThrowError();
  });

  it('hides the close button', () => {
    instance.setState({ inProgress: true });

    const closeButton = root.findAllByProps({ title: 'Close' });
    expect(closeButton).toHaveLength(0);
  });
});

describe('transaction fails', () => {
  const { instance, root } = render(
    <TransactionSender
      onCancel={onCancel}
      onSuccess={onSuccess}
      onFailure={onFailure}
      receiver={publicKey}
      amount={'100.00'}
    />
  );
  instance.setState({ error: new Error() });

  it('renders TransactionFailure', () => {
    expect(() => {
      root.findByType(TransactionFailure);
    }).not.toThrowError();
  });

  it('hides the close button', () => {
    const closeButton = root.findAllByProps({ title: 'Close' });
    expect(closeButton).toHaveLength(0);
  });

  it('passes the correct handler to the subcomponent', () => {
    const subcomponent = root.findByType(TransactionFailure);
    expect(subcomponent.props.onAcknowledge).toEqual(instance.handleFailure);
  });
});

describe('transaction succeeds', () => {
  const { instance, root } = render(
    <TransactionSender
      onCancel={onCancel}
      onSuccess={onSuccess}
      onFailure={onFailure}
      receiver={publicKey}
      amount={'100.00'}
    />
  );
  instance.setState({ response: fakeResponse });

  it('renders TransactionSuccess when response was received', () => {
    expect(() => {
      root.findByType(TransactionSuccess);
    }).not.toThrowError();
  });

  it('hides the close button', () => {
    const closeButton = root.findAllByProps({ title: 'Close' });
    expect(closeButton).toHaveLength(0);
  });

  it('passes the correct handler to the subcomponent', () => {
    const subcomponent = root.findByType(TransactionSuccess);
    expect(subcomponent.props.onAcknowledge).toEqual(instance.handleSuccess);
  });
});
