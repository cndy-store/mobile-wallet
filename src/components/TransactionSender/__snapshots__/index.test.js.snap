// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly without an error 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "height": 1334,
      },
      Object {
        "backgroundColor": "#ffffff",
      },
    ]
  }
  virtual={undefined}
>
  <View
    onLayout={[Function]}
  >
    <View
      style={
        Object {
          "backgroundColor": "#F8F8F8",
          "borderBottomColor": "#a7a6ab",
          "borderBottomWidth": 0.5,
          "elevation": 3,
          "flexDirection": "row",
          "height": 64,
          "justifyContent": "center",
          "left": 0,
          "paddingLeft": 6,
          "paddingRight": 10,
          "paddingTop": 18,
          "right": 0,
          "shadowColor": undefined,
          "shadowOffset": undefined,
          "shadowOpacity": undefined,
          "shadowRadius": undefined,
          "top": 0,
        }
      }
      virtual={undefined}
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "alignSelf": "center",
            "flex": 1,
          }
        }
        virtual={undefined}
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": null,
              "backgroundColor": "transparent",
              "borderBottomWidth": null,
              "borderColor": null,
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "elevation": 0,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "space-between",
              "marginLeft": -3,
              "opacity": 1,
              "paddingBottom": 6,
              "paddingLeft": 4,
              "paddingRight": 6,
              "paddingTop": 6,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#007aff",
                "fontFamily": "System",
                "fontSize": 17,
                "marginLeft": 0,
                "marginRight": 0,
                "paddingLeft": 2,
                "paddingRight": undefined,
                "top": 1,
              }
            }
            uppercase={false}
            virtual={undefined}
          >
            Cancel
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "flex": 1,
          }
        }
        virtual={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={1}
          style={
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 17,
              "fontWeight": "700",
              "marginLeft": undefined,
              "paddingLeft": 4,
              "paddingTop": 1,
              "textAlign": "center",
            }
          }
          virtual={undefined}
        >
          Send CNDY
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "alignSelf": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-end",
          }
        }
        virtual={undefined}
      />
    </View>
  </View>
  <RCTScrollView
    automaticallyAdjustContentInsets={false}
    contentContainerStyle={
      Array [
        Object {
          "padding": 10,
        },
        undefined,
      ]
    }
    contentInset={
      Object {
        "bottom": 0,
      }
    }
    enableAutomaticScroll={true}
    enableResetScrollToCoords={true}
    extraHeight={75}
    extraScrollHeight={0}
    getScrollResponder={[Function]}
    handleOnScroll={[Function]}
    keyboardDismissMode="interactive"
    keyboardOpeningTime={250}
    keyboardShouldPersistTaps="handled"
    keyboardSpace={0}
    onScroll={[Function]}
    padder={true}
    resetKeyboardSpace={[Function]}
    resetScrollToCoords={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    scrollEventThrottle={1}
    scrollForExtraHeightOnAndroid={[Function]}
    scrollToEnd={[Function]}
    scrollToFocusedInput={[Function]}
    scrollToPosition={[Function]}
    showsVerticalScrollIndicator={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
    viewIsInsideTabBar={false}
    virtual={undefined}
  >
    <View>
      <View>
        <View
          style={
            Object {
              "backgroundColor": "#fff",
              "borderBottomWidth": 0.5,
              "borderColor": "#ccc",
              "borderLeftWidth": 0.5,
              "borderRadius": 2,
              "borderRightWidth": 0.5,
              "borderTopWidth": 0.5,
              "elevation": 3,
              "flex": 1,
              "flexWrap": "nowrap",
              "marginBottom": 5,
              "marginLeft": 2,
              "marginRight": 2,
              "marginTop": 5,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.1,
              "shadowRadius": 1.5,
            }
          }
          virtual={undefined}
        >
          <View
            header={true}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderBottomWidth": null,
                "borderRadius": 2,
                "flexDirection": "row",
                "paddingBottom": 15,
                "paddingLeft": 15,
                "paddingRight": 15,
                "paddingTop": 15,
              }
            }
            virtual={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": "600",
                }
              }
              uppercase={false}
              virtual={undefined}
            >
              Enter Public Key of Receiver
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 2,
                "flexDirection": "row",
                "paddingBottom": 10,
                "paddingLeft": 15,
                "paddingRight": 15,
                "paddingTop": 10,
              }
            }
            virtual={undefined}
          >
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                style={Object {}}
                virtual={undefined}
              >
                <View
                  accessibilityComponentType={undefined}
                  accessibilityLabel={undefined}
                  accessibilityTraits={undefined}
                  accessible={true}
                  collapsable={undefined}
                  hasTVPreferredFocus={undefined}
                  hitSlop={undefined}
                  isTVSelectable={true}
                  nativeID={undefined}
                  onLayout={undefined}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "borderBottomWidth": 1,
                      "borderColor": "#D9D5DC",
                      "borderLeftWidth": 0,
                      "borderRightWidth": 0,
                      "borderTopWidth": 0,
                      "flexDirection": null,
                      "marginLeft": 15,
                      "minHeight": 65,
                      "opacity": 1,
                    }
                  }
                  testID={undefined}
                  tvParallaxProperties={undefined}
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Object {
                        "alignSelf": "flex-start",
                        "color": "#575757",
                        "fontSize": 15,
                        "left": null,
                        "paddingLeft": null,
                        "paddingRight": 5,
                        "paddingTop": 5,
                        "position": null,
                        "right": null,
                        "top": 5,
                      }
                    }
                    virtual={undefined}
                  >
                    Receiver
                  </Text>
                  <TextInput
                    allowFontScaling={true}
                    autoCorrect={false}
                    autoFocus={true}
                    editable={true}
                    onChangeText={[Function]}
                    placeholderTextColor="#575757"
                    style={
                      Object {
                        "alignSelf": "stretch",
                        "color": "#000",
                        "flex": 1,
                        "fontSize": 17,
                        "height": 50,
                        "lineHeight": 18,
                        "marginLeft": null,
                        "paddingLeft": null,
                        "paddingRight": 5,
                        "top": 1.5,
                        "width": null,
                      }
                    }
                    underlineColorAndroid="rgba(0,0,0,0)"
                    value=""
                    virtual={undefined}
                  />
                </View>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#fff",
                "borderRadius": 2,
                "flexDirection": "row",
                "paddingBottom": 10,
                "paddingLeft": 15,
                "paddingRight": 15,
                "paddingTop": 10,
              }
            }
            virtual={undefined}
          >
            <View
              style={
                Object {
                  "alignItems": "flex-start",
                  "alignSelf": "stretch",
                  "flex": 1,
                }
              }
              virtual={undefined}
            >
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignItems": "center",
                    "alignSelf": "stretch",
                    "backgroundColor": "#007aff",
                    "borderBottomWidth": null,
                    "borderColor": "#007aff",
                    "borderLeftWidth": null,
                    "borderRadius": 5,
                    "borderRightWidth": null,
                    "borderTopWidth": null,
                    "elevation": 2,
                    "flexDirection": "row",
                    "height": 45,
                    "justifyContent": "center",
                    "opacity": 1,
                    "paddingBottom": 6,
                    "paddingTop": 6,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#fff",
                      "fontFamily": "System",
                      "fontSize": 16.5,
                      "marginLeft": 0,
                      "marginRight": 0,
                      "paddingLeft": 16,
                      "paddingRight": 16,
                    }
                  }
                  uppercase={false}
                  virtual={undefined}
                >
                  Next
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
